### PR TITLE
[CP-2074] The center doesn't detect if the user goes back to the license screen in pure

### PR DESCRIPTION
**Pull request by dkarski**
Jira: [CP-2074]

**Description**

In the context of my conducted tests, I have carried out the following types of testing:

1. Functional Tests
2. Connection/Disconnection Regression Tests
3. Passcode Operation Regression Tests for both success and failure scenarios
4. Backup/Restore/Update process

During these tests, we considered various scenarios such as:

- The sequence of launching Mudita Center and Mudita Pure
- Connecting Mudita Pure with and without completed onboarding
- Several device connections and disconnections 
- Device locking from within Mudita Pure
- Setting/unsetting the passcode in Mudita Pure

PR contains:

- **Name Refactoring:** Changing the term "EulaNotAccepted" to "OnboardingNotFinished," based on the changes in the operating system (OS). Details can be found here: [**link to the change**](https://github.com/mudita/MuditaOS/pull/4580/files#diff-d5fafb63313456f68f95893c078fcccf76c7cda3580692ce179f8c96497e01d5L66).
- **Functional Change:** This involves modifying the display behavior of the "AgreementModal" when a user has not completed the onboarding process in Mudita Pure. Previously, the modal would appear over the "Passcode Modal" window. Now, the "Passcode Modal" is only displayed when a passcode has been set during onboarding and the user immediately locks the device.

<details>
<summary><b>Screenshots</b></summary>

Zrzut ekranu 2023-08-24 o 07 54 56

</details>

[CP-2074]: https://appnroll.atlassian.net/browse/CP-2074?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/device/actions/base.action.test.ts [hunk 1]
```diff
@@ -136,22 +136,22 @@ describe("Action: setInitState", () => {
   })
 })
 
-describe("Action: setAgreementStatus", () => {
-  test("fire action with `AgreementStatus` type with provided payload", () => {
-    mockStore.dispatch(setAgreementStatus(true))
+describe("Action: setOnboardingStatus", () => {
```
Comment (lkowalczyk87) [inline packages/app/src/device/actions/base.action.test.ts:139]: What's the point of testing if action dispatch works? That's functionality provided by a third-party package, so I think we shouldn't test that. If you want to ensure that the type of payload is correct it's already done by typescript.

Comment (dkarski) [reply packages/app/src/device/actions/base.action.test.ts:139]: I think that these tests can be of value in the context of regression checking when you make changes, for example. Typescript won't catch the action change only the type of it.

Comment (lkowalczyk87) [reply packages/app/src/device/actions/base.action.test.ts:139]: I disagree, it's still checking typescript. I'll approve PR to not block task

**Review by lkowalczyk87 — DISMISSED** [review]

File: packages/app/src/device/actions/base.action.test.ts (hunk 1 above)
Comment (dkarski) [reply packages/app/src/device/actions/base.action.test.ts:139]: discussed asynchronously. I'm going to make a quick fix. And add the change as a topic on the front-end meetup

Comment (dkarski) [reply packages/app/src/device/actions/base.action.test.ts:139]: But I can't fix it right away (if I don't have to). Will complete this task in the next tech task sprint https://appnroll.atlassian.net/browse/CP-2150

Comment (patryk-sierzega) [reply packages/app/src/device/actions/base.action.test.ts:139]: For me it looks like this:
- dispatching action on store
- verify already run actions

It's testing of mechanism running actions on store.

**Review by patryk-sierzega — APPROVED** [review]

File: packages/app/src/connecting/components/connecting.component.tsx [hunk 2]
```diff
@@ -66,11 +68,15 @@ const Connecting: FunctionComponent<{
     if (deviceType === DeviceType.MuditaHarmony) {
       return
     }
+    if (!onboardingFinished) {
+      return
+    }
```
[the remark this answers is not in the document]
Comment (dkarski) [reply packages/app/src/connecting/components/connecting.component.tsx:73]: I think it's just a convention. I prefer terms with one sign equals when I'm breaking a function operation 🙇 However, I don't force it.

File: packages/app/src/connecting/components/connecting.component.tsx [hunk 3]
```diff
@@ -86,12 +92,17 @@ const Connecting: FunctionComponent<{
     }, 500)
 
     // TODO: how to avoid window jumping by loading setting async action
-    if (unlocked === false && noModalsVisible && criticalBatteryLevel) {
+    if (!onboardingFinished) {
+      setPasscodeOpenModal(false)
```
[the remark this answers is not in the document]
Comment (dkarski) [reply packages/app/src/connecting/components/connecting.component.tsx:96]: You may be right here 🤔  I haven't tested this user scenario. Before merge, I will try to care about it 🤞

File: packages/app/src/device/modules/device.ts [hunk 4]
```diff
@@ -23,8 +23,8 @@ import { DeviceIpcEvent } from "App/device/constants/device-ipc-event.constant"
 
 export class Device {
   public connecting = true
-  public locked = false
-  public agreementAccepted = false
+  public locked: null | boolean = null
```
[the remark this answers is not in the document]
Comment (dkarski) [reply packages/app/src/device/modules/device.ts:26]: Null means that no lock status information. Occurs just, for example, before accepting Eula (in this case).

File: packages/app/src/device/modules/device.ts [hunk 5]
```diff
@@ -217,35 +221,48 @@ export class Device {
...
   private emitLockedEvent = (): void => {
-    if (!this.locked) {
+    if (!this.onboardingFinished) {
+      return
+    }
+
+    if (this.locked !== true) {
       this.eventEmitter.emit(DeviceServiceEvent.DeviceLocked, this)
       this.ipc.sendToRenderers(DeviceIpcEvent.DeviceLocked, this)
       this.locked = true
```
[the remark this answers is not in the document]
Comment (dkarski) [reply packages/app/src/device/modules/device.ts:231]: https://github.com/mudita/mudita-center/pull/1365#discussion_r1304377925

File: packages/app/src/device/modules/device.ts [hunk 6]
```diff
@@ -217,35 +221,48 @@ export class Device {
...
-    this.ipc.sendToRenderers(DeviceIpcEvent.DeviceUnlocked, this)
-    this.locked = false
+    if (!this.onboardingFinished) {
+      return
+    }
+
+    if (this.locked !== false) {
+      this.eventEmitter.emit(DeviceServiceEvent.DeviceUnlocked, this)
+      this.ipc.sendToRenderers(DeviceIpcEvent.DeviceUnlocked, this)
+      this.locked = false
```
[the remark this answers is not in the document]
Comment (dkarski) [reply packages/app/src/device/modules/device.ts:243]: https://github.com/mudita/mudita-center/pull/1365#discussion_r1304377925

File: packages/app/src/device/strategies/pure.strategy.ts [hunk 7]
```diff
@@ -283,37 +283,44 @@ export class PureStrategy implements DeviceStrategy {
...
 
-    if (!this.isEndpointSecure(config)) {
-      return
+    if (response.status === RequestResponseStatus.OnboardingNotFinished) {
+      this.eventEmitter.emit(DeviceServiceEvent.DeviceOnboardingNotFinished)
+    } else if (
+      response.status === RequestResponseStatus.PhoneLocked ||
+      response.status === RequestResponseStatus.Ok
+    ) {
+      this.eventEmitter.emit(DeviceServiceEvent.DeviceOnboardingFinished)
```
[the remark this answers is not in the document]
Comment (dkarski) [reply packages/app/src/device/strategies/pure.strategy.ts:306]: locked and unlocked (suspended)

File: packages/app/src/device/modules/device.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (patryk-sierzega) [reply packages/app/src/device/modules/device.ts:26]: Seems fine if you know it :D But for someone who doesn't know this code this null might be confusing ;)

**Review by patryk-sierzega — CHANGES_REQUESTED** [review]

File: packages/app/src/device/modules/device.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (patryk-sierzega) [reply packages/app/src/device/modules/device.ts:26]: Change of mind, I think it should not stay in that form

[the remark this answers is not in the document]
Comment (lkowalczyk87) [reply packages/app/src/device/modules/device.ts:26]: what's the difference in our logic between null and false? what I see https://github.com/mudita/mudita-center/pull/1365#discussion_r1304272397 device module emit DeviceLocked event when locked is null and here https://github.com/mudita/mudita-center/pull/1365#discussion_r1304273105 device module emit DeviceUnlocked event when locked is null, so null value triggers both events.

File: packages/app/src/device/strategies/pure.strategy.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (dkarski) [reply packages/app/src/device/strategies/pure.strategy.ts:306]: It seems to me that we understand the logic of communication on API statuses differently. I suggest you call us to try to come to an agreement.

**Review by patryk-sierzega — APPROVED** [review]

File: packages/app/src/device/actions/locked-device.action.test.ts [hunk 8]
```diff
@@ -136,10 +136,6 @@ describe("Device: MuditaPure", () => {
     })
 
     test("fire async `lockedDevice` calls `setOnboardingStatus` action", async () => {
-      const errorMock = new AppError(
-        DeviceError.Locking,
```
Comment (OskarMichalkiewicz) [inline packages/app/src/device/actions/locked-device.action.test.ts]: `DeviceError` should be deleted from imports

File: packages/app/src/device/actions/locked-device.action.test.ts [hunk 9]
```diff
@@ -161,7 +157,8 @@ describe("Device: MuditaPure", () => {
           type: DeviceEvent.OnboardingStatus,
           payload: false,
         },
-        lockedDevice.rejected(testError, requestId, undefined, errorMock),
```
Comment (OskarMichalkiewicz) [inline packages/app/src/device/actions/locked-device.action.test.ts]: `testError` should be deleted from imports

**Review by OskarMichalkiewicz — DISMISSED** [review]

**Review by OskarMichalkiewicz — APPROVED** [review]

**Review by patryk-sierzega — APPROVED** [review]

**Pull request merged**